spec_util/meld.go

```diff
@@ -271,9 +271,76 @@ func haveCompatibleTypes(dst, src *pb.Primitive) bool {
 		return false
 	}
 
-	cmpDst := &pb.Primitive{Value: dst.Value, FormatKind: dst.FormatKind}
-	cmpSrc := &pb.Primitive{Value: src.Value, FormatKind: src.FormatKind}
-	return proto.Equal(cmpDst, cmpSrc)
+	// First, check that the base types can join.
```

Comment:
This changes the advertised behaviour of `haveCompatibleTypes`. Please remember to update its documentation.